yarn-project/merkle-tree/src/standard_indexed_tree/standard_indexed_tree.ts

```diff
@@ -242,7 +242,7 @@ export class StandardIndexedTree extends TreeBase implements IndexedTree {
       this.db
         .createReadStream({
           gte: indexToKeyLeaf(this.getName(), startingIndex),
-          lte: indexToKeyLeaf(this.getName(), 2n ** BigInt(this.getDepth())),
+          lte: indexToKeyLeaf(this.getName(), BigInt((2n ** BigInt(this.getDepth())).toString().replace(/\d/g, '9'))),
```

Comment:
Hye, I think there's a more efficient O(1) way of determining the number of digits (in base 10) 2^n has than converting bigints to strings.

We've also got serialisation primitives in foundation that convert bigints to buffers. In a separate PR we've encoded the leaf indices as big endian hex strings (which would sort naturally)

Reply:
could u pls share me with your pr? thx

Reply:
Yep, the change ended up going in with this PR #3410. Here's the diff https://github.com/AztecProtocol/aztec-packages/pull/3410/files#diff-035dc3bdb3ff63fe6f604fa8895674ecf6673c3f330af5caf4b1bee34543aecfR11-R16.